code/window_snap.py

```diff
@@ -0,0 +1,221 @@
+"""Tools for voice-driven window management.
+
+Originally from dweil/talon_community - modified for newapi by jcaw.
+
+"""
+
+# TODO: Map keyboard shortcuts to this manager once Talon has key hooks on all
+#   platforms
+
+import time
+from operator import xor
+from typing import Optional
+
+from talon import ui, Module, Context, actions
+
+
+def sorted_screens():
+    """Return screens sorted by their topmost, then leftmost, edge.
+
+    Screens will be sorted left-to-right, then top-to-bottom as a tiebreak.
+
+    """
+
+    return sorted(
+        sorted(ui.screens(), key=lambda screen: screen.visible_rect.top),
+        key=lambda screen: screen.visible_rect.left,
+    )
+
+
+def _set_window_pos(window, x, y, width, height):
+    """Helper to set the window position."""
+    # TODO: Special case for full screen move - use os-native maximize, rather
+    #   than setting the position?
+    rect = ui.Rect(round(x), round(y), round(width), round(height))
+    window.rect = rect
+    # HACK: If on Windows and the window was maximised, this can leave the
+    #   window in a weird semi-maximised state. Some windows will be tiny or
+    #   invisible in this state. Repeating seems to solve this.
+    actions.sleep("250ms")
```

Comment:
Just curious,  is this long of a delay necessary? On windows 10 here, seems to work without a delay, but haven't tested under load or anything.

Reply:
The original had a similar hack. 

https://github.com/dwiel/talon_community/blob/master/misc/window_snap.py#L58

You could probably get away with shorter on Windows, but I copied that in case it needed to be reintroduced for Mac. The original method set the rect three times - not sure if that would still be necessary.

Reply:
(To be clear, testing on this end, it does need a small delay to be stable on Windows.)

Reply:
Maybe the delay should be gated with a if windows block similar to the mac specific hack?

Reply:
I think that's reasonable.